core/src/main/java/org/infinispan/util/CacheCollectionMapper.java

```diff
@@ -0,0 +1,51 @@
+package org.infinispan.util;
+
+import org.infinispan.CacheCollection;
+import org.infinispan.CacheStream;
+import org.infinispan.commons.util.CloseableIterator;
+import org.infinispan.commons.util.CloseableIteratorMapper;
+import org.infinispan.commons.util.CloseableSpliterator;
+import org.infinispan.commons.util.CloseableSpliteratorMapper;
+
+import java.util.function.Function;
+
+/**
+ * A {@link CacheCollection} that allows for a different set to be mapped as
+ * a different instance with values replaced on request.  This is useful as a
+ * cache collection that is normally lazily evaluated to prevent having to
+ * pull all values into memory which can be a lot faster when checking single
+ * values and can also prevent out of memory issues.
+ *
+ * @param <E> Type of elements in collection before transformation
+ * @param <R> Type of elements in collection after transformation
+ * @since 9.0
+ */
+public class CacheCollectionMapper<E, R> extends CollectionMapper<E, R> implements CacheCollection<R> {
```

Comment:
I realized I missed this one in the previous PR later last night, but I figured you had it under control :)